Contributed upstream in https://github.com/Dao-AILab/flash-attention/pull/958

commit df984badb88a4e5332615792a17247eef56a2194
Author: Doug Hellmann <dhellmann@redhat.com>
Date:   Fri May 17 09:01:48 2024 -0400

    add pyproject.toml with build dependencies
    
    Adding this metadata means it is possible to `pip install flash-attn`
    without pre-installing the packages imported in setup.py. It is still
    possible to follow the existing manual instructions, too, but by not
    requiring pre-installation it _also_ makes it easier for someone to
    build from the source dist.
    
    Signed-off-by: Doug Hellmann <dhellmann@redhat.com>

diff --git a/pyproject.toml b/pyproject.toml
new file mode 100644
index 0000000..dc450d8
--- /dev/null
+++ b/pyproject.toml
@@ -0,0 +1,3 @@
+[build-system]
+requires = ["setuptools", "packaging", "psutil", "ninja", "torch", "wheel"]
+build-backend = "setuptools.build_meta"
\ No newline at end of file
